### DISCUSSION TITLE
How many threads are in the thread pool?

### DISCUSSION THREAD

**Discussion by temeddix** [fzyzcjy/flutter_rust_bridge#1007] [Q&A] [answered]
Hi, I would like to thank you first for this great package. I've been playing with Flutter Rust Bridge for a while.

First of all, I got curious about how many threads are generated when Flutter Rust Bridge is used. I did some test for myself which `println!()`s the current thread ID in Rust and found out there are 4 threads in the pool. However, my PC has 16 logical cores and I was wondering if that number of 4 is fixed. Can I decrease that number to 1 if I don't need multiple number of threads in the pool? Because more threads means more stack memory, and although stack memory allocation for 3 more threads won't be that big, it will still be a waste of resource.

Also, the docs say that I can [even throw away the thread pool](https://cjycode.com/flutter_rust_bridge/feature/concurrency.html?highlight=thread#concurrency). How can I do that? I couldn't find it in the docs.

**Comment by fzyzcjy** [comment, marked as answer]
You are welcome!

> First of all, I got curious about how many threads are generated when Flutter Rust Bridge is used.

https://cjycode.com/flutter_rust_bridge/feature/handler.html
You can create custom handler, and thus customize #threads (or even use your own implementation etc)

The default one https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/frb_rust/src/handler.rs uses https://github.com/fzyzcjy/flutter_rust_bridge/blob/master/frb_rust/src/thread.rs thus 4 threads.
